src/index.ts

```diff
@@ -1,11 +1,122 @@
-export interface ImplementMe {
-	myFunc(): Promise<string>;
+import type { BalenaPricing, Credit } from './types';
+
+// Credit pricing definitions.
+// Initially restrict credit usage to device:microservices feature.
+export const CREDITS: { [key: string]: Credit } = {
+	'device:microservices': {
+		basePrice: 150,
+		firstDiscountPrice: 149,
+		discountRate: 0.33,
+		discountThreshold: 12000,
+		discountThresholdPrice: 125,
+	},
+};
+
+/**
+ * Gets and returns pricing for a given feature
+ * @param featureSlug - feature slug
+ * @returns credit pricing parameters
+ *
+ * @example
+ * const pricing = getFeaturePricing({...}, 'device:microservices');
+ */
+function getFeaturePricing(
+	credits: { [key: string]: Credit },
+	featureSlug: string,
+): Credit {
+	const pricing = credits[featureSlug];
+	if (pricing == null) {
+		throw new Error('Requested feature not allowed for credit usage');
+	}
+	return pricing;
 }
 
-const moo = 1;
+class Pricing implements BalenaPricing {
+	public credits: { [key: string]: Credit };
+
+	public constructor(credits = CREDITS) {
+		this.credits = credits;
+	}
+
+	public getCreditPrice(
+		featureSlug: string,
+		availableCredits: number,
+		creditsToPurchase: number,
+	): number {
+		// Assert that credit amounts are valid
+		if (availableCredits < 0) {
+			throw new Error('Available credits must be >= 0');
+		}
+		if (creditsToPurchase < 0) {
+			throw new Error('Credit purchase amount must be >= 0');
+		}
+
+		const pricing = getFeaturePricing(this.credits, featureSlug);
+		const total = availableCredits + creditsToPurchase;
+		if (creditsToPurchase === 0 || total === 0) {
+			return 0;
+		} else if (total <= pricing.discountThreshold) {
+			return Math.round(
+				pricing.firstDiscountPrice +
+					((pricing.discountThresholdPrice - pricing.firstDiscountPrice) /
+						(pricing.discountThreshold - 1)) *
+						(total - 1),
+			);
+		} else {
+			return Math.round(
+				pricing.discountThresholdPrice *
+					Math.pow(
+						1 - pricing.discountRate,
+						Math.log10(total / pricing.discountThreshold),
+					),
+			);
+		}
+	}
+
+	public getCreditTotalPrice(
+		featureSlug: string,
+		availableCredits: number,
+		creditsToPurchase: number,
+		additionalDiscount: number,
```

Comment:
The `additionalDiscount`:
- can be a positive or a negative
- is applied on the result of the curve calculation
- is only used in `getCreditTotalPrice()` and `getTotalSavings()` as it is applied to totals and not individual credit unit prices

Reply:
I would actually argue that the `additionalDiscount` could also be removed out of this package, since that should only be an exception to our pricing that the CS team should do rarely-if ever. The UI & marketing site will not be using them, which sounds like it could become an API-only thing, especially given that all that it does is `Math.round(X * (1 - additionalDiscount))`.
@rhampt ?

Reply:
To be honest, it's only here as I just copied this over from my previous balena-api PR as-is. I think it makes sense to remove from here as well.